core/evtforward/ethcall/result.go

```diff
@@ -0,0 +1,67 @@
+package ethcall
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"code.vegaprotocol.io/vega/core/types"
+	"github.com/PaesslerAG/jsonpath"
+)
+
+type Result struct {
+	Bytes         []byte
+	Values        []any
+	Normalised    map[string]string
+	PassesFilters bool
+}
+
+func NewResult(spec types.EthCallSpec, bytes []byte) (Result, error) {
```

Comment:
nit-pick, but I think it makes more sense to return `*Result`, and return `nil` in case of errors to force the caller to check the error

Reply:
Well I intentionally wanted Result to be 'value' type; according to my understanding of the best practice (it's in the google style guide somewhere) the [genreal advice](https://go.dev/tour/methods/8) seems to be to use a value type unless you've got methods that need to modify the instance, or if you worried about the cost of copying it.

Conceptually the Result is an immutable thing, and it isn't huge, that's what I went for.